overlord/ifacestate/helpers.go

```diff
@@ -295,3 +295,33 @@ func getConns(st *state.State) (map[string]connState, error) {
 func setConns(st *state.State, conns map[string]connState) {
 	st.Set("conns", conns)
 }
+
+// CheckInterfaces checks whether plugs and slots of snap are allowed for installation.
+func CheckInterfaces(st *state.State, snap *snap.Info) error {
```

Comment:
CheckInterfaces() is a pretty generic name. I wonder if CheckInstallationInterfaces() or something else would be better.